metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_schema.py

```diff
@@ -27,6 +28,17 @@ class SnowflakeFK:
     referred_column_names: List[str]
 
 
+@dataclass
+class SnowflakeTag:
+    database: str
+    schema: str
+    name: str
+    value: str
+
+    def __str__(self):
+        return f"{self.database}.{self.schema}.{self.name}={self.value}".lower()
```

Comment:
The format of the stringification can be discussed.

Reply:
Hey @frsann Let's change this string conversion to use `key:value` form. Also Is there any  explicit need to convert this to lowercase ? I would recommend keeping these to their original case.

Reply:
Fixed. I removed the `lower` and replaced it with calling the `self.snowflake_identifier` function which lowercases based on the config.

Reply:
Okay, that seems fair. We should keep display name [tagProperties->name] unlowercased - in exactly same case as snowflake, as we do for snowflake tables, schema, etc.

Reply:
Also, I noticed that snowflake tags can have descriptions (for key part)

`create tag cost_center comment = 'cost_center tag';`

Would it make sense to add it to tag description ?

Reply:
We could get the comment, but it would of course require an extra `show tags` or `select * from snowflake.account_usage.tags`-query and complicate the tag-workunit creation slightly. It would also give us the `allowed_value` but, it might not be super interesting in Datahub. Let me know if you want it added.   

I fixed the un-lowercasing of the tag name now, only the urn is optionally lowercased.